### ISSUE TITLE
.markdownlint.json in project root seemingly is being ignored

### ISSUE THREAD

**Issue by kilasuit** [super-linter/super-linter#5101]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Errors are being raised in builds like [this one](https://github.com/PowerShell/PowerShell/actions/runs/7442657230/job/20246410950?pr=20991) from [this PR](https://github.com/PowerShell/PowerShell/pull/20991) as the projects `.markdownlint.json` file that is in the root of the project is being ignored & as such errors are being raised when it shouldn't be.

### Expected Behavior

file in project root should be respected as this is used in editor tooling too so makes no sense not to respect the settings enforced as part of the project.

We set this in our .markdownlint.json
```json
"MD007": {
        "indent": 4
    },
```
which should not then cause any errors in builds

### Super-Linter version

```markdown
super-linter/super-linter@v5
```

### Relevant log output

```shell
2024-01-08 03:02:12 [INFO]   File:[/github/workspace/docs/community/working-group-definitions.md]
2024-01-08 03:02:12 [ERROR]   Found errors in [markdownlint] linter!
2024-01-08 03:02:12 [ERROR]   Error code: 1. Command output:
------
/github/workspace/docs/community/working-group-definitions.md:52:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:53:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:175:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:176:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:177:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:179:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:180:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:189:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:190:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:192:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
/github/workspace/docs/community/working-group-definitions.md:193:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
```

### Steps To Reproduce

Above info should be more than enough to be able to reproduce

**Comment by ferrarimarco** [comment]
Hi @kilasuit !

I think this is a misconfiguration, rather than an issue. By looking at your [workflow definition](https://github.com/PowerShell/PowerShell/blob/master/.github/workflows/markdownLink.yml), you didn't set `LINTER_RULES_PATH`, so super-linter picks its default value: `.github/linters`. The result is that super-lint expects that linter configuration files are in that directory. For example, Markdownlint config file path should be: `.github/linters/.markdownlint.json`

At this point, I think you have two options:

- Either you move `.markdownlint.json` to the `.github/linters` directory, or
- Set `LINTER_RULES_PATH` to `.` (the root of your repository in the `env` section of the `Lint Markdown` step in `.github/workflows/markdownLink.yml`

Finally, to improve performance a bit when upgrading to the next super-linter release, you may want to replace `FILTER_REGEX_INCLUDE: .*\.md` with `VALIDATE_MARKDOWN: true` in `.github/workflows/markdownLink.yml`. This is not necessary to solve your issue tho.

Let me know if this works for you :)

**Comment by ferrarimarco** [comment]
PS: If you want to read the full story, this is documented in the README: https://github.com/super-linter/super-linter?tab=readme-ov-file#configure-super-linter

**Comment by kilasuit** [comment]
@ferrarimarco I've tried setting `LINTER_RULES_PATH` to `.` as well as `MARKDOWN_CONFIG_FILE : .markdownlint.json` and a few other variations, but this still was failing. I can't spend more time on this right now and for what we want we may just directly use the [mardownlint-cli2-action](https://github.com/marketplace/actions/markdownlint-cli2-action) instead, but isn't my call to make on that project.

I still think that superlinter should auto look for `.markdownlint.json` anywhere in the repo and use it & would be a good update in a future release.

**Comment by ferrarimarco** [comment]
> @ferrarimarco I've tried setting LINTER_RULES_PATH to . as well as MARKDOWN_CONFIG_FILE : .markdownlint.json and a few other variations, but this still was failing. I can't spend more time on this right now and for what we want we may just directly use the [mardownlint-cli2-action](https://github.com/marketplace/actions/markdownlint-cli2-action) instead, but isn't my call to make on that project.

@kilasuit Ah, that's more surprising. Can you point me to the workflow run where this did happen so I can have a look at the logs?

> I still think that superlinter should auto look for .markdownlint.json anywhere in the repo and use it & would be a good update in a future release.

Repositories can be quite big to scan, and users might set arbitrary names for config files. Also, they might have several config files in the repo for the same linter to accomodate for different use cases.

**Comment by kilasuit** [comment]
There were a few different runs before I reverted to the current in use workflow definition which you can look through with [this query](https://github.com/PowerShell/PowerShell/actions/workflows/markdownLink.yml?query=actor%3Akilasuit+branch%3Apatch-3)

**Comment by ferrarimarco** [comment]
Ah, I see what the issue is by looking at the workflow definition files for all those runs. There's no workflow file that sets the two variables that you need:

- `LINTER_RULES_PATH: .`
- `MARKDOWN_CONFIG_FILE : .markdownlint.json`

I think it should work if you add those to the `Lint Markdown` step

**Comment by ferrarimarco** [comment]
So, I would try to refactor that step to look like this:

```yaml
    - name: Lint Markdown
      uses: super-linter/super-linter@v5
      env:
        VALIDATE_ALL_CODEBASE: false
        DEFAULT_BRANCH: master
        VALIDATE_MARKDOWN: true
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        LINTER_RULES_PATH : .
```

(there is no need to set `VALIDATE_EDITORCONFIG: false` because you set `VALIDATE_MARKDOWN: true`, as described at the end of the [Configure super-linter section in the README](https://github.com/super-linter/super-linter?tab=readme-ov-file#configure-super-linter).)

**Comment by kilasuit** [comment]
Thanks, missed this yesterday but will try this out in the current PR I am working on now
